packages/sdk-communication-layer/src/services/MessageHandlers/onCommunicationLayerMessage.ts

```diff
@@ -0,0 +1,78 @@
+import { RemoteCommunication } from '../../RemoteCommunication';
+import { CommunicationLayerMessage } from '../../types/CommunicationLayerMessage';
+import { EventType } from '../../types/EventType';
+import { MessageType } from '../../types/MessageType';
+import { handleAuthorizedMessage } from './handleAuthorizedMessage';
+import { handleOriginatorInfoMessage } from './handleOriginatorInfoMessage';
+import { handleOtpMessage } from './handleOtpMessage';
+import { handlePauseMessage } from './handlePauseMessage';
+import { handleReadyMessage } from './handleReadyMessage';
+import { handleTerminateMessage } from './handleTerminateMessage';
+import { handleWalletInfoMessage } from './handleWalletInfoMessage';
+
+/**
+ * Central dispatcher function to handle messages for a `RemoteCommunication` instance.
+ *
+ * The function takes a message of type `CommunicationLayerMessage` and a `RemoteCommunication` instance
+ * and based on the `message.type` decides which specific handler function should be invoked.
+ *
+ * Steps taken by the function:
+ *
+ * 1. Logs the incoming message if `debug` mode is enabled.
+ * 2. Sets the `ready` status of the instance to `true`.
+ * 3. Checks the `message.type` and the `isOriginator` status of the instance to determine the relevant handler.
+ * 4. Invokes the specific handler function based on the determined conditions.
+ * 5. If the message doesn't match specific criteria, it emits a general `MESSAGE` event.
+ *
+ * @param message The incoming `CommunicationLayerMessage` that needs to be processed.
+ * @param instance The `RemoteCommunication` instance that is the target of the message.
+ */
+export function onCommunicationLayerMessage(
+  message: CommunicationLayerMessage,
+  instance: RemoteCommunication,
+) {
+  if (instance.state.debug) {
+    console.debug(
+      `RemoteCommunication::${
+        instance.state.context
+      }::on 'message' typeof=${typeof message}`,
+      message,
+    );
+  }
+
+  instance.state.ready = true;
+
+  if (
```

Comment:
what about
```
type MessageHandler = (instance: any, message?: any) => void;

const handlers: Record<MessageType, MessageHandler | undefined> = {
  [MessageType.ORIGINATOR_INFO]: !instance.state.isOriginator ? (i, m) => handleOriginatorInfoMessage(i, m) : undefined,
  [MessageType.WALLET_INFO]: instance.state.isOriginator ? (i, m) => handleWalletInfoMessage(i, m) : undefined,
  [MessageType.TERMINATE]: handleTerminateMessage,
  [MessageType.PAUSE]: handlePauseMessage,
  [MessageType.READY]: instance.state.isOriginator ? handleReadyMessage : undefined,
  [MessageType.OTP]: instance.state.isOriginator ? (i, m) => handleOtpMessage(i, m) : undefined,
  [MessageType.AUTHORIZED]: instance.state.isOriginator ? handleAuthorizedMessage : undefined,
};

const handler = handlers[message.type];
if (handler) {
  handler(instance, message);
  return;
}

```

Reply:
We don't `return` in every condition, only in `MessageType.ORIGINATOR_INFO` and `MessageType.WALLET_INFO`
The rest of the condtions would cause the function to continue to the last line which is,
`  instance.emit(EventType.MESSAGE, message);`
So we need to think if we want to return now in any case, and when we want to call to
`  instance.emit(EventType.MESSAGE, message);`

What do you think?

Reply:
ok let's keep it like that for now and we'll refactor this if needed later when we'll have tests implemented to check that everything is woking correctly ;)